sql/indexer_internal_commands_optimized.sql

```diff
@@ -0,0 +1,153 @@
+WITH
+  coinbase_receiver_info AS (
+    SELECT
+      bic.block_id,
+      bic.internal_command_id,
+      bic.sequence_no,
+      bic.secondary_sequence_no,
+      coinbase_receiver_pk.value AS coinbase_receiver
+    FROM
+      blocks_internal_commands AS bic
+      INNER JOIN internal_commands AS ic ON bic.internal_command_id=ic.id
+      INNER JOIN blocks_internal_commands AS bic_coinbase_receiver ON bic.block_id=bic_coinbase_receiver.block_id
+      AND (
+        bic.internal_command_id<>bic_coinbase_receiver.internal_command_id
+        OR bic.sequence_no<>bic_coinbase_receiver.sequence_no
+        OR bic.secondary_sequence_no<>bic_coinbase_receiver.secondary_sequence_no
+      )
+      INNER JOIN internal_commands AS ic_coinbase_receiver ON ic.command_type='fee_transfer_via_coinbase'
+      AND ic_coinbase_receiver.command_type='coinbase'
+      AND bic_coinbase_receiver.internal_command_id=ic_coinbase_receiver.id
+      INNER JOIN public_keys AS coinbase_receiver_pk ON ic_coinbase_receiver.receiver_id=coinbase_receiver_pk.id
+  ),
+  internal_commands_info AS (
+    SELECT DISTINCT
+      ON (
+        bic.block_id,
+        bic.internal_command_id,
+        bic.sequence_no,
+        bic.secondary_sequence_no
+      ) i.id,
+      i.command_type AS "command_type: InternalCommandType",
+      i.receiver_id,
+      i.fee,
+      i.hash,
+      pk.value AS receiver,
+      cri.coinbase_receiver AS "coinbase_receiver?",
+      bic.sequence_no,
+      bic.secondary_sequence_no,
+      bic.block_id,
+      bic.status AS "status: TransactionStatus",
+      b.state_hash,
+      b.height
+    FROM
+      internal_commands AS i
+      INNER JOIN blocks_internal_commands AS bic ON i.id=bic.internal_command_id
+      INNER JOIN public_keys AS pk ON i.receiver_id=pk.id
+      INNER JOIN blocks AS b ON bic.block_id=b.id
+      AND (
+        b.chain_status='canonical'
+        OR b.chain_status='pending'
+      )
```

Comment:
It seems that adding this while dropping the following CTEs:

```
  canonical_blocks AS (
    SELECT
      *
    FROM
      blocks
    WHERE
      chain_status='canonical'
  ),
  max_canonical_height AS (
    SELECT
      max(HEIGHT) AS max_height
    FROM
      canonical_blocks
  ),
  pending_blocks AS (
    SELECT
      b.*
    FROM
      blocks AS b,
      max_canonical_height AS m
    WHERE
      b.height>m.max_height
      AND b.chain_status='pending'
  ),
  blocks AS (
    SELECT
      *
    FROM
      canonical_blocks
    UNION ALL
    SELECT
      *
    FROM
      pending_blocks
  )
```
improves performance by another ~50%... updated the performance tests table in PR summary.